### PR TITLE
Marc's stuff

### PR DESCRIPTION
**Pull request by nandezer**
I made the class of the WritingStyleAnalyzer, I called WAnalyzerS.java.

Functions made: getSentenceLengthProbabilities and getWordLengthProbabilities.

### CODE REVIEW

**Comment by carlthome** [comment]
1. We're actually not using GitHub Flow anymore but in this case I appreciate it. :+1: 
2. The class you've written is pretty messy. You have dead code, weird naming conventions and also changed the interface to perform extraneous IO. I do not support this. Please do general improvements both coding style-wise and functionality-wise. If you have a method to calculate a probability density function of misspellings then you also should construct a humanize function that takes the function and a text as input and returns a purposefully misspelt text.

**Comment by nandezer** [comment]
I did the general improvements both coding style-wise and functionality-wise that you asked

**Comment by carlthome** [comment]
Cool. Checking!

**Comment by carlthome** [comment]
You're still reading the entire file on every method call. That seems insanely wasteful. You're also mixing naming conventions for variables. In general the code just seems like a very roundabout way of implementing the class. 200 lines, really? Please do general improvements. Sorry.

**Comment by nandezer** [comment]
Tried to optimize the code by reading the text on the first function only (analyze) and getting how many words, sentences and paragraphs has.
Also changed a some variables names.

**Comment by nandezer** [comment]
Humanizer part mainly finish.
(I want to add an algorithm that depending the probability of misspelling words choose one word or an other; as higher the probability less similarity between words)

Also changed the misspelling probability calculation from words per sentences to words per paragraph since when I humanize the text I read it per paragraphs.

**Comment by carlthome** [comment]
@johnPertoft and @nandezer, you both need a function that picks one element from a set based on a probabilitiy distribution. Refactor the code into a generalized method for this purpose. A new class with a static method would work.

**Comment by nandezer** [comment]
Humanizing part done (works fine) it's a bit poor the algorithm to select the words though.

File: src/main/java/com/github/ai14/prosammgen/Humanizer.java [hunk 1]
```diff
@@ -1,5 +1,159 @@
...
+    }
+
+
+    /**
+     * Calculates similarities between two words (the possible misspelling word and the original one)
+     * @param WordsToCompare
+     * @param OriginalWord
+     * @return
+     */
+    private double[] getSimilaritudesBetweenWords(List<String> WordsToCompare, String OriginalWord) {
```
Comment (carlthome) [inline src/main/java/com/github/ai14/prosammgen/Humanizer.java]: What's a similaritude?

File: src/main/java/com/github/ai14/prosammgen/Humanizer.java [hunk 2]
```diff
@@ -1,5 +1,159 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.*;
+import java.util.Random;
+import java.util.function.Function;
+
 public class Humanizer {
-  // TODO https://github.com/ai14/prosammgen/issues/17
+
+    private List<String> Words;
```
Comment (carlthome) [inline src/main/java/com/github/ai14/prosammgen/Humanizer.java]: Please keep to using camelCase. You're mixing weird conventions. Read: https://en.wikipedia.org/wiki/Naming_convention_%28programming%29#Java and keep to that.

File: src/main/java/com/github/ai14/prosammgen/Humanizer.java [hunk 3]
```diff
@@ -1,5 +1,159 @@
...
+    public Humanizer()throws IOException, ParseException {
+        //Load words (both correct and misspelled ones)
+        this.Words = Files.readAllLines(Paths.get("res/words"));
+        WAnalyzerS Analyze = new WAnalyzerS();
+        Analyze.analyze(Paths.get("example/previousreflectiondocument.in"));
+        this.misspellingProbability = Analyze.getMisspellingWordsProbabilities();
+    }
+
+    /**
+     * * This "humanize" the text by making some misspelled words given a text, a misspellingProb (as higher the probability
```
Comment (carlthome) [inline src/main/java/com/github/ai14/prosammgen/Humanizer.java]: Weird english. Fix.

File: src/main/java/com/github/ai14/prosammgen/Humanizer.java [hunk 4]
```diff
@@ -1,5 +1,159 @@
...
+        for(int i = 0; i < WordsToCompare.size(); ++i){
+            ratingMisspeled[i] = StringUtils.getJaroWinklerDistance(WordsToCompare.get(i), OriginalWord);
+        }
+        return ratingMisspeled;
+    }
+
+
+
+    public   List<String> checkForPossibleMisspellingWords(String correctWord){
+        boolean ChapitalLetter = false;
```
Comment (carlthome) [inline src/main/java/com/github/ai14/prosammgen/Humanizer.java]: What's a chapital?

File: src/main/java/com/github/ai14/prosammgen/Humanizer.java [hunk 5]
```diff
@@ -1,5 +1,159 @@
...
+        numberOfParagraphs = paragraph.length;
+        System.err.println(numberOfParagraphs);
+        StringBuilder HumanizedText = new StringBuilder();
+        //each paragraph
+        for (int i = 0; i < numberOfParagraphs;++i){
+            int numberOfWords = 0;
+            String[] WordsParagraph = paragraph[i].split("\\s+");
+            numberOfWords = WordsParagraph.length;
+            //each word
+            for (int j =0; j < numberOfWords; ++j) {
```
Comment (carlthome) [inline src/main/java/com/github/ai14/prosammgen/Humanizer.java:44]: Weird formatting.

File: src/main/java/com/github/ai14/prosammgen/Humanizer.java [hunk 6]
```diff
@@ -1,5 +1,159 @@
...
+     */
+    public String TextHumanizer(String text)throws IOException, ParseException {
+        //TODO: Change frequency of misspelling a word
+        int numberOfParagraphs = 0;
+        String[] paragraph = text.split("[\\n\\n]+");
+        numberOfParagraphs = paragraph.length;
+        System.err.println(numberOfParagraphs);
+        StringBuilder HumanizedText = new StringBuilder();
+        //each paragraph
+        for (int i = 0; i < numberOfParagraphs;++i){
```
Comment (carlthome) [inline src/main/java/com/github/ai14/prosammgen/Humanizer.java]: Why not just skip the comment and do a foreach loop instead?

File: src/main/java/com/github/ai14/prosammgen/Humanizer.java [hunk 7]
```diff
@@ -1,5 +1,159 @@
...
+                        newWord = chooseWord(possibleMisspellingWords, SimilarityRate);
+                    }
+                    //TODO: take of the print
+                    System.err.println(WordsParagraph[j]+"   "+newWord);
+                }
+                HumanizedText.append(newWord+" ");
+            }
+            HumanizedText.append("\n\n");
+        }
+        HumanizedText.append("\\end{document}");
```
Comment (carlthome) [inline src/main/java/com/github/ai14/prosammgen/Humanizer.java]: What's this? This seems incredibly badly thought out. The humanizer should not perform LaTeX formatting at all!

**Comment by carlthome** [comment]
Still seems kinda shaky, everything. Needs general improvements. I think you know yourself when the code is getting mature enough for a merge into master. Please rework it up to that point. Also, don't forget to merge master into your branch as soon as David's additions have been merged into master.

**Comment by carlthome** [comment]
Since I've stared at this pull request for quite some time now I'm losing sight of the important stuff. @johnPertoft, could you take over when Marc needs a code review next time?

**Pull request merged**